packages/plugin-coverage/src/lib/coverage-plugin.ts

```diff
@@ -34,37 +25,30 @@ export const RUNNER_OUTPUT_PATH = join(
  *
  * @returns Plugin configuration.
  */
-export function coveragePlugin(config: CoveragePluginConfig): PluginConfig {
-  const { reports, perfectScoreThreshold, coverageTypes, coverageToolCommand } =
-    coveragePluginConfigSchema.parse(config);
+export async function coveragePlugin(
+  config: CoveragePluginConfig,
+): Promise<PluginConfig> {
+  const coverageConfig = coveragePluginConfigSchema.parse(config);
 
-  const audits = coverageTypes.map(
+  const audits = coverageConfig.coverageTypes.map(
     (type): Audit => ({
       slug: `${type}-coverage`,
       title: `${capitalize(type)} coverage`,
-      description: `${capitalize(type)} coverage percentage on the project.`,
+      description: coverageDescription[type],
     }),
   );
 
-  const getAuditOutputs = async () =>
-    perfectScoreThreshold
-      ? applyMaxScoreAboveThreshold(
-          await lcovResultsToAuditOutputs(reports, coverageTypes),
-          perfectScoreThreshold,
-        )
-      : await lcovResultsToAuditOutputs(reports, coverageTypes);
+  const group: Group = {
+    slug: 'coverage',
+    title: 'Code coverage metrics',
+    description: 'Group containing all defined coverage types as audits.',
+    refs: audits.map(audit => ({ ...audit, weight: 1 })),
```

Comment:
Should they all be given same weights? :thinking:

Reply:
I suppose I could give a bigger weight to function coverage, otherwise unless the user sets it up differently, I thought that error vs warning distinction was enough. WDYT?

Reply:
Errors vs warnings is good for distinguishing issues, but it doesn't have impact on category score. If function coverage should have more impact than lets say line coverage, then it would be cool to reflect that in the score calculation.

Reply:
Alright, will think of some.